geopy/geocoders/geonames.py

```diff
@@ -171,13 +189,67 @@ def reverse(
             'lng': lng,
             'username': self.username
         }
-        url = "?".join((self.api_reverse, urlencode(params)))
+        url = "?".join((self.nearby_api, urlencode(params))) if find_nearby \
```

Comment:
Just changing the url is not enough. These API endpoints accept different sets of parameters, so the arguments passed to this function should be validated for each API endpoint independently, and the incompatible ones should cause a `ValueError` being raised.